.editorconfig

```diff
@@ -0,0 +1,162 @@
+# EditorConfig is awesome:http://EditorConfig.org
+
+# top-most EditorConfig file
+root = true
+
+# Don't use tabs for indentation.
+[*]
+indent_style = space
+# (Please don't specify an indent_size here; that has too many unintended consequences.)
+
+# Code files
+[*.cs]
+indent_size = 4
+insert_final_newline = true
+charset = utf-8-bom
+
+# 'new' expression can be simplified (requires C# v9.0+)
+dotnet_diagnostic.IDE0090.severity = none
+
+# Xml project files
+[*.{csproj,vbproj,vcxproj,vcxproj.filters,proj,projitems,shproj}]
+indent_size = 2
+
+# Xml config files
+[*.{props,targets,ruleset,config,nuspec,resx,vsixmanifest,vsct}]
+indent_size = 2
+
+# JSON files
+[*.json]
+indent_size = 2
+
+# Dotnet code style settings:
+[*.cs]
+# Sort using and Import directives with System.* appearing first
+dotnet_sort_system_directives_first = true
+# Avoid "this." and "Me." if not necessary
+dotnet_style_qualification_for_field = false:suggestion
+dotnet_style_qualification_for_property = false:suggestion
+dotnet_style_qualification_for_method = false:suggestion
+dotnet_style_qualification_for_event = false:suggestion
+
+# Use language keywords instead of framework type names for type references
+dotnet_style_predefined_type_for_locals_parameters_members = true:suggestion
+dotnet_style_predefined_type_for_member_access = true:suggestion
+
+# Suggest more modern language features when available
+dotnet_style_object_initializer = true:suggestion
+dotnet_style_collection_initializer = true:suggestion
+dotnet_style_coalesce_expression = true:suggestion
+dotnet_style_null_propagation = true:suggestion
+dotnet_style_explicit_tuple_names = true:suggestion
+
+# Non-private static fields are camelCase
+dotnet_naming_rule.non_private_static_fields_should_be_camel_case.severity = suggestion
+dotnet_naming_rule.non_private_static_fields_should_be_camel_case.symbols = non_private_static_fields
+dotnet_naming_rule.non_private_static_fields_should_be_camel_case.style = non_private_static_field_style
+
+dotnet_naming_symbols.non_private_static_fields.applicable_kinds = field
+dotnet_naming_symbols.non_private_static_fields.applicable_accessibilities = public, protected, internal, protected internal, private protected
+dotnet_naming_symbols.non_private_static_fields.required_modifiers = static
+
+dotnet_naming_style.non_private_static_field_style.capitalization = camel_case
+
+# Constants are PascalCase
+dotnet_naming_rule.constants_should_be_pascal_case.severity = suggestion
+dotnet_naming_rule.constants_should_be_pascal_case.symbols = constants
+dotnet_naming_rule.constants_should_be_pascal_case.style = constant_style
+
+dotnet_naming_symbols.constants.applicable_kinds = field, local
+dotnet_naming_symbols.constants.required_modifiers = const
+
+dotnet_naming_style.constant_style.capitalization = pascal_case
+
+# Static fields are camelCase and start with s_
```

Comment:
From your experience, are all these prefix rules a part of an established convention, or is the current code written like that? Or is it something else?

Reply:
Hm... I must have missed/never encountered static fields.

After searching I see YAFC does use `static` but withsome some prefix (e.g. here for `Instance` and `listPad`):
https://github.com/have-fun-was-taken/yafc-ce/blob/b0fa78f6b4e91e6b572ec8eec00c5c9ae950bed1/YAFC/Windows/DependencyExplorer.cs#L12-L16

Unfortunately both with camelCase and PascalCase... :disappointed: 

And preference for when I fix the `s_` prefix in the config?

Reply:
On a (quick) glance, it looks like cameCase is winning.
`Instance` is written in _all_ cases with PascalCase though... So there seems to be some logic in this. But I am afraid we cannot have both in the config

Reply:
I removed the `s_` and kept camelCase

Reply:
Just to be clear, by "all these prefix rules" I meant all rules about prefixes. But I'll need to check the code to see if the existing code uses them.

Reply:
Oh, sorry I misunderstood/misread.

> From your experience, are all these prefix rules a part of an established convention, or is the current code written like that? Or is it something else?

The current code is written like this. Although it is not matching/adhering 100%. For example `static readonly MyClass Instance`, which is used all over the place, is not camelCase...

Reply:
[Surprisingly](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/identifier-names), prefixes are a part of the established coding conventions. Then, I suggest to keep them. If Microsoft suggests this code style, it means that most of the potential C# contributors are familiar with it.

Reply:
> we need to reformat the whole code-base

I consider this a legacy-code situation. You do the actual changes you need, and then fix the code style of whatever you consider necessary. Usually, it's the main file of the change, but it can be more.

In other words, it's not mandatory to reformat anything right after the merge, unless some of the rules mark it as a problem. If that's the case, then this rule will need to be updated to be a warning instead.

Reply:
I found these defaults: https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/code-style-rule-options#example-editorconfig-file

So I (blindly, did not check with our codebase) put them in the editorfile, so we adhere to the standards of .net/Microsoft.

Reply:
Thank you! I was in the middle of going through them to essentially suggest what you did :D
I'll go through them to see what should be changed from `silent` to `suggestion`.